File: python-algorithm/ds_7_array_inplace/L667-m.cpp
//
// Created by 李恒 on 2020/6/29.
//
#include <iostream>
#include <vector>

using namespace std;
class Solution {
public:
    vector<int> constructArray(int n, int k) {
        vector<int> ret(n,0) ;
        ret[0] = 1;
        for (int i=1, interval=k; i<=k; i++, interval--) {
            ret[i] = i%2 == 1 ? ret[i-1] + interval : ret[i-1] - interval;
        }
        for (int i = k + 1; i < n; i++) {
            ret[i] = i + 1;
        }
        return ret;
    }
};

File: python-algorithm/ds_7_array_inplace/L667-m.py
# -*- coding: utf-8 -*-
# @Time    : 2020/10/5 10:39
# @Author  : No Name

class Solution:
    def constructArray(self, n: int, k: int) :
        res = [1 for i in range(n)]
        i = 1
        interval = k
        while i <= k:
            if i%2 == 1:
                res[i] = res[i-1]+interval
            else:
                res[i] = res[i-1]-interval
            i += 1
            interval -= 1
        for i in range(k+1, n):
            res[i] = i+1
        return res

res = Solution().constructArray(10,5)
print(res)